This epatch serves the below:
1. Invoke building of BuildTools used during cross-compiling.
2. Restrict the compilation only till Buildtools and exit without
   compiling the entire llvm package.

fix by Shirish S <shirish.s@amd.com>

--- llvm-3.8.1.orig/Makefile	2016-12-08 10:05:10.426645891 +0530
+++ llvm-3.8.1.src/Makefile	2016-12-08 10:23:16.280134853 +0530
@@ -101,6 +101,7 @@
 # If we're cross-compiling, build the build-hosted tools first
 ifeq ($(LLVM_CROSS_COMPILING),1)
+buildtools-only:: cross-compile-build-tools
 all:: cross-compile-build-tools
 
 clean::
 	$(Verb) rm -rf BuildTools
@@ -156,7 +157,7 @@
 # running the dist-check target
 DIST_CHECK_CONFIG_OPTIONS = --with-llvmgccdir=$(LLVMGCCDIR)
 
-.PHONY: debug-opt-prof
+.PHONY: llvm-buildtools-only
 debug-opt-prof:
 	$(Echo) Building Debug Version
 	$(Verb) $(MAKE)
@@ -178,6 +179,7 @@
 clang-only: all
 tools-only: all
 libs-only: all
+llvm-buildtools-only: buildtools-only
 install-clang: install
 install-libs: install
 
